### ISSUE TITLE
(skyewm): can we combine the backprop loop counter and main gradient 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#80]
loop into a single loop? The original Python code doesn't combine the
loops, but I'm not sure why.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [301b14c240fe99249dc2225132a7ebe5cbecbdc4](https://github.com/imdone/tensorflow/commit/301b14c240fe99249dc2225132a7ebe5cbecbdc4) by Skye Wanderman-Milne skyewm@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/cc/framework/while_gradients.cc#L186-L193
